Code/src/main/java/nl/utwente/group10/ui/components/ConnectionStateDependent.java

```diff
@@ -0,0 +1,49 @@
+package nl.utwente.group10.ui.components;
+
+import nl.utwente.group10.ui.handlers.ConnectionCreationManager;
+
+/**
+ * Interface for Nodes who are dependent on their ConnectionState.
+ */
+public interface ConnectionStateDependent {    
+    /**
+     * Tells the Node that its current state (considering connections) possibly has
+     * changed. Default implementation does not react to a potential state
+     * change.
+     *
+     * This method should only be called after the Node's constructor is done.
+     * 
+     * This method will invalidate the Node even if the state did not change.
+     */
+    public void invalidateConnectionState();
+    
+    /**
+     * @return The ConnectionState the Node is in.
+     */
+    public int getConnectionState();
+    
+    /**
+     * Does the same as invalidateConnectionState(), but cascading down to other
```

Comment:
> down

Which direction is down? And are you sure those are the only nodes you have to invalidate?